Utilisation de variables d'environnement pour faciliter la configuration

lfavole opened the pull request:
Bonjour @solene-drnx,

J'ai déplacé toute la configuration dans des variables d'environnement.
Cela veut dire que tu n'as plus besoin de modifier le code pour changer la liste des cinémas ou pour indiquer ta clé API Mapbox.
Tu ne risqueras donc plus de committer ta clé API lorsque tu modifies le code.

J'ai aussi ajouté une variable `WEBSITE_TITLE` pour changer le titre du site, comme cela a été fait pour Brest.

Il te suffit de créer un fichier `.env` et d'y mettre ta configuration, par exemple :
```
MAPBOX_TOKEN=pk.xxxxxxxxxx
WEBSITE_TITLE=CinéBrest
```

Chacun pourra donc facilement créer sa propre instance avec les cinémas de sa ville.

lfavole committed:
Add env variables for configuration

lfavole committed:
Always load default env file

solene-drnx commented on app.py:
Coucouu, c'est super pratique, je connaissais pas mais pour ne pas push la clé d'API c'est pas mal ahah ! 

J'viens de tester, est ce qu'il faudrait pas rajouter `dotenv.load_dotenv(".env")` juste après la ligne 11 dans app.py ? pour bien charger le fichier `.env`

lfavole replied:
Non, pas besoin, [Flask le fait automatiquement](https://flask.palletsprojects.com/en/stable/api/#flask.Flask.run) si `python-dotenv` est installé.
~~Par contre je pense que la ligne 11 va remplacer les variables d'environnement déjà définies (dans l'environnement, pas dans le fichier `.env`) donc il faudra que je corrige cela.~~
Pas besoin, il y a l'option [`override=False`](https://saurabh-kumar.com/python-dotenv/reference/) qui empêche déjà cela.

solene-drnx replied:
mais je crois que même si y'a le override=False, faut quand même charger explicitement le .env avant sinon ca va juste pas le charger initialement

```python
dotenv.load_dotenv(dotenv_path=".env")
dotenv.load_dotenv(dotenv_path=".env.sample", override=False)
```

lfavole committed:
Always load the .env file

If the app is run in another way than `python app.py`, the environment file will not be loaded.

lfavole commented:
En fait il faut toujours charger le fichier `.env` parce que c'est la fonction `app.run()` qui charge automatiquement les variables d'environnement.
Si on lance l'application autrement ([par exemple sur un serveur](https://stackoverflow.com/q/58409892#comment139331441_58416673)), ça ne marchera pas.
[(Voir ici aussi)](https://flask.palletsprojects.com/en/stable/cli/#environment-variables-from-dotenv)

solene-drnx commented:
Let's go, merci beaucoup Laurent 🔥
